### ISSUE TITLE
MultiMeshInstance2D can cull early

### ISSUE THREAD

**Issue by reikken** [godotengine/godot#54854]
### Godot version

3.4.stable

### System information

Windows10, GLES3

### Issue description

I encountered this for MultiMeshInstance2Ds generated at runtime.
This node can be culled from drawing before all instances are off screen. It seems to cull as soon as the center of the node is off screen. 

Resizing the window seems to fix the issue for any existing instances, and ones generated entirely in a _ready() function or premade in a packed scene do not seem to have this issue.

### Steps to reproduce

Create a new multimesh but do not set the instances until some amount of time after adding it to the scene tree. Then pan the camera until the multimesh is halfway offscreen, and it will disappear.

### Minimal reproduction project

[MM2DTEST.zip](https://github.com/godotengine/godot/files/7516544/MM2DTEST.zip)

**Comment by Calinou** [comment]
For reference, this issue was cross-posted on the Godot Q&A: https://godotengine.org/qa/120419/multimeshinstance2d-culling-before-going-fully-offscreen

**Comment by reikken** [comment]
It does seem like it is being calculated correctly when it does get calculated, but isn't always up to date

**Comment by kleonc** [comment]
> If we can't calculate its visibility rect automatically, it's possible that MultiMeshInstance2D may need a property to set its visibility rect manually (like GPU-based Particles2D).

For `MultiMeshInstance` (3D) [`GeometryInstance::set_custom_aabb()`](https://docs.godotengine.org/en/3.4/classes/class_geometryinstance.html#class-geometryinstance-method-set-custom-aabb) can be used.
For `MultiMeshInstance2D` seems like there's indeed no counterpart method allowing to set custom rect. But it should be doable using [`VisualServer::canvas_item_set_custom_rect()`](https://docs.godotengine.org/en/3.4/classes/class_visualserver.html#class-visualserver-method-canvas-item-set-custom-rect):
```
var mmi: MultiMeshInstance2D = ...
var rect: Rect2 = ...
VisualServer.canvas_item_set_custom_rect(mmi.get_canvas_item(), true, rect)
```

Also it seems that `MultiMesh::get_aabb()` [recalculates its AABB if it's dirty](https://github.com/godotengine/godot/blob/0529813cf2bb4f92c55d19807961ef3606b3c054/drivers/gles3/rasterizer_storage_gles3.cpp#L4916-L4923) (in fact it triggers updating AABBs of all dirty multimeshes). So as of now this should work to manually update `MultiMeshInstance2D`'s visibility rect:
```
var mmi: MultiMeshInstance2D = ...
var aabb: AABB = mmi.multimesh.get_aabb()
var rect := Rect2(aabb.position.x, aabb.position.y, aabb.size.x, aabb.size.y)
VisualServer.canvas_item_set_custom_rect(mmi.get_canvas_item(), true, rect)
```

**Comment by akien-mga** [comment]
Fixed by #54899.
